File: skysail.restlet/src/io/skysail/restlet/menu/MenuItemDescriptor.java
package io.skysail.restlet.menu;

import io.skysail.domain.Entity;
import io.skysail.domain.html.Field;
import io.skysail.restlet.forms.ListView;

public class MenuItemDescriptor implements Entity {

    @Field
    @ListView
    private String url;

    private String name;

    public MenuItemDescriptor(MenuItem menuItem) {
//        name = menuItem.get;
//        String link = menuItem.getLink();
//        url = "<a href='"+link+"'>"+name+"</a>";
    }

    @Override
    public String getId() {
        return url;
    }


}

File: skysail.restlet/src/io/skysail/restlet/menu/MenuItemDescriptor.scala
package io.skysail.restlet.menu

import io.skysail.domain.ddd.ScalaEntity
import io.skysail.core.html.Field
import io.skysail.restlet.forms.ListView

class MenuItemDescriptor(menuItem: MenuItem) extends ScalaEntity[String] {

  @Field
  @ListView(hide = true)
  val url = "<a href='" + link + "'>" + name + "</a>";

  val name = menuItem.name
  val link = menuItem.link

  var id: Option[String] = Some(url)

}
